src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -19,6 +20,121 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class Binder
     {
+#nullable enable
+        /// <summary>
+        /// This is an argument that needs to be considered for MAMM rules. It's at least an 
+        /// output argument of the function and very often also an input. Essentially this 
+        /// represents all the in, ref and out arguments.
+        /// </summary>
+        internal readonly struct MixableArgument
+        {
+            internal BoundExpression Argument { get; }
+
+            /// <summary>
+            /// In the case this is the argument for a ref / out parameter this will refer
+            /// to the corresponding parameter. This will be null in cases like arguments 
+            /// passed to an arglist
+            /// </summary>
+            internal ParameterSymbol? Parameter { get; }
+
+            internal bool IsOut => Parameter is { RefKind: RefKind.Out };
+
+            internal MixableArgument(ParameterSymbol? parameter, BoundExpression argument)
+            {
+                Argument = argument;
+                Parameter = parameter;
+            }
+
+            public override string? ToString() => Parameter is { } p
+                ? p.ToString()
+                : Argument.ToString();
+        }
+
+        /// <summary>
+        /// Represents an argument being analyzed for escape analysis purposes. This represents the
+        /// argument as written. For example a `ref x` will only be represented by a single 
+        /// <see cref="EscapeArgument"/>
+        /// </summary>
+        internal readonly struct EscapeArgument
+        {
+            /// <summary>
+            /// This will be null in cases like arglist or a function pointer receiver.
+            /// </summary>
+            internal ParameterSymbol? Parameter { get; }
+
+            internal BoundExpression Argument { get; }
+
+            internal RefKind RefKind { get; }
+
+            internal bool IsArgList { get; }
+
+            internal EscapeArgument(ParameterSymbol? parameter, BoundExpression argument, RefKind refKind, bool isArgList = false)
+            {
+                Debug.Assert(!isArgList || parameter is null);
+                Argument = argument;
+                Parameter = parameter;
+                RefKind = refKind;
+                IsArgList = isArgList;
```

Comment:
It looks like this property is never read

Reply:
Correct. Will remove.